src/H5private.h

```diff
@@ -2473,6 +2478,16 @@ H5_DLL herr_t H5CX_pop(hbool_t update_dxpl_props);
   #define HDcompile_assert(e)     do { typedef struct { unsigned int b: (e); } x; } while(0)
 */
 
+/* Private typedefs */
+
+/* Union for const/non-const pointer for use by functions that manipulate
+ * pointers but do not write to their targets or return pointers to const
+ * specified locations.  This helps us avoid compiler warnings. */
+typedef union {
+    void *      vp;
+    const void *cvp;
+} H5_flexible_const_ptr_t;
```

Comment:
I am not super excited about this hack. Adding code complexity to paper over warning seems like a really bad idea. How much does the warning count change if this is removed?

Reply:
I outlined what I viewed as the options in a comment here a while back. Jordan and I discussed this and decided this was the least offensive solution. I understand the linux kernel does something similar. I believe we could get the warnings down to 1 or 2 by just casting away const, though that's not great. With this solution, instead of just ignoring const I've at least given guidelines on when this union can and can't be used.